geoprocessing/src/modules/files/files.service.ts

```diff
@@ -2,32 +2,41 @@ import { Injectable } from '@nestjs/common';
 import { createReadStream } from 'fs';
 import { Extract } from 'unzipper';
 import * as path from 'path';
-import { unlink, rmdir } from 'fs/promises';
+import { unlink, rmdir, readdir } from 'fs/promises';
 
 @Injectable()
 export class FileService {
-  unzipFile(fileInfo: Express.Multer.File): Promise<string> {
+  unzipFile(
+    origin: string,
+    fileName: string,
+    destination: string,
+  ): Promise<string> {
     return new Promise((resolve, reject) => {
-      createReadStream(fileInfo.path)
+      createReadStream(origin)
         .pipe(
           Extract({
             path: path.join(
-              fileInfo.destination,
-              path.basename(fileInfo.filename.replace('.zip', '')),
+              destination,
+              path.basename(fileName.replace('.zip', '')),
```

Comment:
Hmm, actually
```suggestion
              path.basename(fileName, '.zip'),
```
will do the same

Reply:
Thanks @Dyostiq 3891b195524fcb5a64e83d62f0ca8b010c42dbba